NN_module/precision.f90

```diff
@@ -0,0 +1,13 @@
+module precision
+
+  use, intrinsic :: iso_fortran_env, only : sp=>real32, dp=>real64
+  ! Imports primitives used to interface with C
+  use, intrinsic :: iso_c_binding, only: c_sp=>c_float, c_dp=>c_double
+
+  implicit none
+
+  public
+  integer, parameter :: c_wp = c_sp
+  integer, parameter :: wp = 4
```

Comment:
Shouldn't this technically be `real32`?  (which is indeed `4` usually!) but I assume that was why the iso_fortran_env was imported.

Reply:
Although I guess as the code we are testing uses just `4` this is okay?

Reply:
Yeah, ideally it would be real32, but I set it to this for consistency with the rest of the SAM code.
However, perhaps we want to update it to be consistent with CAM, which I think is r8?

Reply:
It depends what we are testing here: do we want to test it in "SAM mode" (kind 4) or "CAM mode" (kind 8) Or both? Does that have implications for the tolerances in test_utils? (although the tolerance looks relatively high so probably not).

Reply:
Latest commit updates precision to have a cam and sam precision, adds sp and dp options to the test utils routines, and swaps 4 for real32. I know some systems _can_ have a difference between these two but generally not an issue on modern machines I believe.